### DISCUSSION TITLE
Cannot Install a Pure Python Package

### DISCUSSION THREAD

**Discussion by 11301858** [pyscript/pyscript#1371] [Q&A] [answered]
Hello, 

I am using pyscript for my project. I would like to import the XAISuite python library: https://pypi.org/project/XAISuite/#files

This library has a pure python wheel. However, when I try to include it by using "py-config", I get the following error:

![Screen Shot 2023-04-08 at 6 47 28 PM](https://user-images.githubusercontent.com/66180831/230750078-f5cccad9-f6a7-4285-b10e-ea6146bf0e93.png)

Could someone please let me know how I can use XAISuite? I have looked in many forums like Stack Overflow and Github, but cannot seem to understand why XAISuite cannot be installed. 

Thanks in advance for your help.

**Comment by hoodmane** [comment]
This is an instance of #1308. The problem is that it transitively depends on torch. The actual error from micropip is: `Can't find a pure Python 3 wheel for 'torch>=2.0.0'`

**Reply by 11301858** [reply]
Thanks @hoodmane Is there any estimate of when this will be fixed?

**Reply by hoodmane** [reply]
When we'll port torch to work with Pyodide? Not any time soon. Most recent update is that ryanking13 tried to do it and made this comment about it:
https://github.com/pyodide/pyodide/issues/1625#issuecomment-1008808994

**Reply by 11301858** [reply]
Oh, ok. I tried using a version of XAISuite before it used Pytorch or Tensorflow, and solely used sk-learn. I was still unsuccessful. Do you think there might be another dependency causing this problem?

Thank you for your help though.

![Screen Shot 2023-04-08 at 9 57 01 PM](https://user-images.githubusercontent.com/66180831/230755188-5fd2f9e0-bf49-4b60-9d42-f929d2149e80.png)

**Reply by hoodmane** [reply]
That version says:
```
Can't find a pure Python 3 wheel for: 'scikit-learn<1.2,>=0.24', 'lime', 'hnswlib>=0.5.0', 'shap>=0.40.0'
```
We have scikit-learn but we vendor v1.2.2 which is not <1.2. You could build and host your own older version. lime is a pure Python package but it doesn't have a wheel on pypi, you can make one and host it. hnswlib is a CMake based C++ library, which is a bit of a hassle but is probably pretty reasonable to port. `shap` uses cuda, so porting it is probably very hard.

**Comment by 11301858** [comment, marked as answer]
Ok, Thanks very much for the clarification. I guess Pyscript can't be used for machine learning explainability. 

This is a really good library, though, and thank you for providing me with the opportunity to use it. For now, I've included a py-repl in my [website](https://11301858.github.io/xaisuiteweb/Home.html) with an error message warning users XAISuite can't be used. 

Thanks again.

**Reply by hoodmane** [reply]
Yeah all the ml stuff is really demanding to build!
